portfolio/src/main/webapp/script.js

```diff
@@ -37,18 +37,23 @@ function login(url){
   window.location = url;
 }
 
-function logout(url){
-  window.location = url;
-}
-
-async function checkLoginStatus(){
+async function setButtonName(){
   const response = await fetch('/login');
   const status = await response.text();
-  
-  if(status.localeCompare("/_ah/login?continue=%2Findex.html")){
-      login(status);
+  let obj = JSON.parse(status);
```

Comment:
The variable obj is never re-assigned, so this should be const.
```suggestion
  const obj = JSON.parse(status);
```